src/components/forms/inputs/InputBase.jsx

```diff
@@ -37,9 +37,14 @@ export default class InputBase extends React.Component {
                 { this.renderInput() }
             </div>
         );
-    }
+    } 
 
     onChange(ev) {
         this.props.onValueChange(this.props.name, ev.target.value);
     }
+
+    onBlur(ev) {
+        this.props.onBlur(ev);
+        this.props.onValueChange(this.props.name, ev.target.value);
```

Comment:
I'm guessing the only reason this is working (if it actually is working?) is because you are calling `onValueChange()` after blur, which feels weird to me. Nothing should have changed on blur (except maybe on the next `render()`).